### PR TITLE
Tabular: Added GPU support for LightGBM, XGBoost, CatBoost, Neural Network Models

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

https://github.com/awslabs/autogluon/issues/263, https://github.com/awslabs/autogluon/issues/894

*Description of changes:*

- Added GPU support for LightGBM, XGBoost, CatBoost, Neural Network Models. This is supported both in normal training and during HPO.

- Fixed major bug in CatBoost that occurs when it gets a perfect validation score causing it to perform much worse.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-896-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-896/2/index.html

File: tabular/src/autogluon/tabular/models/lgb/callbacks.py [hunk 1]
```diff
@@ -173,7 +173,7 @@ def _callback(env):
                              train_loss=best_trainloss[i],
                              best_iter_sofar=best_iter[i] + 1,
                              best_valperf_sofar=best_score[i],
-                             eval_metric=eval_metric,
+                             eval_metric=eval_metric,  # Actually stopping_metric
```
Comment (jwmueller) [inline tabular/src/autogluon/tabular/models/lgb/callbacks.py:176]: comment can be clarified

File: tabular/src/autogluon/tabular/models/lgb/lgb_model.py [hunk 2]
```diff
@@ -87,9 +87,7 @@ def _fit(self, X_train=None, y_train=None, X_val=None, y_val=None, dataset_train
                 # TODO: lightgbm must have a special install to support GPU: https://github.com/Microsoft/LightGBM/tree/master/python-package#build-gpu-version
                 #  Before enabling GPU, we should add code to detect that GPU-enabled version is installed and that a valid GPU exists.
                 #  GPU training heavily alters accuracy, often in a negative manner. We will have to be careful about when to use GPU.
```
Comment (jwmueller) [inline tabular/src/autogluon/tabular/models/lgb/lgb_model.py:89]: you may want to report this to users as warning as well, that they may get better accuracy with CPU (I don't think people would expect that).

**Review by jwmueller — APPROVED** [review]

File: tabular/src/autogluon/tabular/models/lgb/lgb_model.py (hunk 2 above)
Comment (Innixma) [reply tabular/src/autogluon/tabular/models/lgb/lgb_model.py:89]: added logging statements for this in LightGBM and CatBoost

**Comment by szha** [comment]
Job PR-896-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-896/3/index.html
